### ISSUE TITLE
[GFF] Deprecate some of these keyboards

### ISSUE THREAD

**Issue by LornaSIL** [keymanapp/keyboards#2259]
Need to review which of these can be "deprecated" in the `keyboard_info` file.

https://keyman.com/keyboards/gff_awn_7 > https://keyman.com/keyboards/gff_awngi_xamtanga
https://keyman.com/keyboards/gff-awn-powerpack-7 > https://keyman.com/keyboards/gff_awngi_xamtanga
https://keyman.com/keyboards/gff-tir-et-powerpack-7 > https://keyman.com/keyboards/gff_tigrinya_ethiopia
https://keyman.com/keyboards/gff-sgw-powerpack-7 > https://keyman.com/keyboards/gff_gurage and https://keyman.com/keyboards/gff_gurage_legacy

I suspect we can deprecate the above, but not the below (yet):
https://keyman.com/keyboards/gff_bcq_7
https://keyman.com/keyboards/gff-bcq-powerpack-7
https://keyman.com/keyboards/gff-mym-powerpack-7
https://keyman.com/keyboards/gff_mym_7

What do you think @dyacob

**Comment by dyacob** [comment]
@LornaSIL I think this deprecation list is perfect.  

I have a new `gff_ethiopic` that is nearly complete, perhaps ready in a few days or over the next weekend.  It is meant to replace `gff_ethiopic_7` and can also be the replacement for the `bcq` and `mym` keyboards (I don't have any plan to update these).

**Comment by LornaSIL** [comment]
Some of the first ones had already been deprecated. I just hadn't keep up with them.
We'll wait on the gff_ethiopic one to deprecate the others.

**Comment by dyacob** [comment]
With the deprecation, can these sections be removed from the respective `.keyboard_info` files?

```
  "related": {
    "gff_sgw_7": {
      "deprecates": true
    }
```

**Comment by LornaSIL** [comment]
I'm not sure what you are asking. It is that line that actually does the
deprecation. Adding DEPRECATE.md is merely informational when someone looks
at the repo and thinks about updating that keyboard.

**Comment by dyacob** [comment]
I was imagining that if the deprecated keyboard was no longer discoverable via the website (due to deprecation) then the `related` section might be referring to a keyboard (`gff_sgw_7`, etc) that was no longer around to be on the receiving end of a reference. In that case, with the referenced item no longer around, it might be ok to remove the `related` section.

**Comment by LornaSIL** [comment]
Ah, ok. I don't think Marc wants keyboards actually deleted,  but you can
see it's harder to get to and pretty obvious we don't want them using the
deprecated keyboard. If we leave the deprecated keyboard, someone with that
link will find the preferred keyboard.

**Comment by DavidLRowe** [comment]
Marc is following the principle that a keyboard should always be available (never deleted, since someone might possibly need it someday), but can be flagged as deprecated so that people won't start using it unintentionally.

**Comment by mcdurdin** [comment]
The deprecation information in the .keyboard_info file is used by Keyman (or can be in the future) to offer upgrades from deprecated keyboards to the replacement version. You can see for example in https://api.keyman.com/keyboard/gff_amh_7, the following relationship data, which is constructed from the deprecation information in multiple .keyboard_info files:

```json
{
  "related": {
    "gff-amh-7": {
      "deprecates": true
    },
    "gff_amharic": {
      "deprecatedBy": true
    }
  }
}
```

We keep the deprecated keyboards available, mostly for the sake of institutions who roll out a specific version of a keyboard and want to have coordinated management of any upgrades. Often times, an institution will be using a very old keyboard, because the upgrade cost is significant -- retraining required, etc. The deprecated keyboards don't show up on the normal search but can be found in the ['Obsolete Keyboard Search'](https://keyman.com/keyboards?obsolete=1) on keyman.com (also found under the Products/Older Versions page). I think it is important we don't dictate the upgrade timetable for users of the older keyboards, hence we keep the older keyboards available.

Furthermore, older versions of current keyboards are still available on downloads.keyman.com, if you know where to look for them, e.g. https://downloads.keyman.com/keyboards/gff_amharic/. We don't publish the data on keyman.com or api.keyman.com, but it's readily available for those who have the know-how and need for them. (Of course, in most cases nowadays the GitHub repo also has the full source history.)

**Comment by dyacob** [comment]
When I look into the `legacy/gff` directory for these older keyboards, I'm not finding the `.kmn` files.  I have local copies of the `.kmn` files that I could add to the repository in their respective folders if there is interest in preserving them for posterity, etc.

**Comment by LornaSIL** [comment]
If you wish, you could add the kmn files under an `archive` folder, but we
won't update anything in "legacy". Those keyboards either must be enhanced
to move to release or left in legacy as kmp files to die a slow death.
Nothing in legacy gets built, it's just the kmp.

**Comment by dyacob** [comment]
Thanks @LornaSIL , I've added the old files to "archive" folders and created PR #2278 .  Then realized I may have misunderstood the archive approach.  I added the kmn files to "archive" folders under their respective parent folders (like `gff_bcq_7/archive` .  Or were suggesting a single archive folder somewhere else, like `release/gff/archive` ?

**Comment by mcdurdin** [comment]
> Then realized I may have misunderstood the archive approach. I added the kmn files to "archive" folders under their respective parent folders (like `gff_bcq_7/archive` .

Perfect :grin:

**Comment by LornaSIL** [comment]
I made a comment on the PR as I don't like adding a folder called `gff_other_7` since that isn't a keyboard we can offer in the current state.

**Comment by LornaSIL** [comment]
Just for consistency, I'd like us to add a `DEPRECATED.md` file to `gff_ethiopic_7`. It's been deprecated already, but for tidyness it would be good to add that (I don't mind if it's done with other keyboard changes). I can do it sometime unless @dyacob gets to it before me.

**Comment by dyacob** [comment]
Lorna, I moved the content out of gff_other_7 and into respective folders under the `release\gff` , only under the gff keyboard folders, I placed them into existing `extras` folders.  Simply to reduce the number of folders.  This meant creating an `extras` folder for gff_gurage, which should have been there, I realized that a layer creation script was missing, so I've also snuck it into this PR.

**Comment by dyacob** [comment]
Apologies for missing these, I've added the missing `DEPRECATED.md` files.  The `release/gff/gff_geez` keyboard should deprecate any previous GFF keyboard for Ge'ez language. I copied the `DEPRECATE .md` file from the `legacy\gff\gff_gez_7` folder into the `legacy\gff\gff-gez-powerpack-7` folder.

**Comment by DavidLRowe** [comment]
@LornaSIL What more needs to be done on this issue?

**Comment by LornaSIL** [comment]
I'll go through and create a new PR tomorrow.

**Comment by dyacob** [comment]
Lest it be overlooked, the "Amharic and Ethiopic Keyboards" page (https://keyman.com/keyboards/h/amharic/) I think should be revised to reflect the deprecations as part of this update or following.  There could well be links to these keyboards under the website in need of removal. Does the Keyman team have a tool to find them?

**Comment by LornaSIL** [comment]
> https://keyman.com/keyboards/h/amharic/

This is a bit challenging. If I send you the file, will you update it as best you can? The Keyman for Windows seems outdated since all the keyboards now work on macOS and Linux as well.

I'm not sure where the graphics reside though. If we need to update the recommended fonts (since you've been modifying them) I'll need to figure that bit out.

The keyboards are quite dated. Here's a list of your non-deprecated keyboards:
https://keyman.com/keyboards/gff_amharic
https://keyman.com/keyboards/gff_amharic_classic
https://keyman.com/keyboards/gff_awngi_xamtanga
https://keyman.com/keyboards/gff_blin
https://keyman.com/keyboards/gff_ethiopic
https://keyman.com/keyboards/gff_geez
https://keyman.com/keyboards/gff_gurage
https://keyman.com/keyboards/gff_gurage_and_amharic
https://keyman.com/keyboards/gff_gurage_legacy
https://keyman.com/keyboards/gff_harari
https://keyman.com/keyboards/gff_harege_fidelat
https://keyman.com/keyboards/gff_mesobe_fidelat
https://keyman.com/keyboards/gff_tigre
https://keyman.com/keyboards/gff_tigrinya_eritrea
https://keyman.com/keyboards/gff_tigrinya_ethiopia

Would we want them all included?

**Comment by mcdurdin** [comment]
The keyman.com/keyboards/h/amharic page is maintained manually. You can see the source at https://github.com/keymanapp/keyman.com/blob/master/keyboards/h/amharic/index.php and create a pull request there if you feel up to the task! (I'd be thrilled if you did!)

**Comment by mcdurdin** [comment]
> I'm not sure where the graphics reside though.

The graphics can be found at https://github.com/keymanapp/keyman.com/tree/master/cdn/dev, so for example `<img src="<?php echo cdn('img/wait.gif'); ?>" />` refers to https://github.com/keymanapp/keyman.com/blob/master/cdn/dev/img/wait.gif

**Comment by dyacob** [comment]
@mcdurdin and @LornaSIL , I'll give it a try.  Maybe not until the weekend (or next).   My inclination here would be to retrieve a list of Ethiopic keyboards with a query and then present a list in HTML markup.  Then the page could be evergreen in this respect and avoid/reduce the need for manual updates.  I know zilch about any available backend databases or APIs, so that's a barrier for me.

**Comment by mcdurdin** [comment]
The purpose of these landing pages is to try and help users differentiate for the popular language options (which is why there is only a dozen or so of them). So we deliberately "curate" the list of available keyboards because we don't necessarily want to recommend all the keyboards available for a given language -- some of them are specialized, or less well maintained. In some cases we try and provide comparative data too, e.g. see https://keyman.com/keyboards/h/greek.

If we just query the database, we have that already with https://keyman.com/keyboards/languages/am

**Comment by mcdurdin** [comment]
And the flipside of that of course is it requires maintenance, but I think the tradeoff is worthwhile for the small set of languages for which we do it.
